spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceConfiguration.java

```diff
@@ -47,8 +47,13 @@
 public class ChannelBindingServiceConfiguration {
 
 	@Bean
+	// This conditional is intentionally not in an autoconfig (usually a bad idea) because
+	// it is used to detect a ChannelBindingService in the parent context (which we know
+	// already exists).
 	@ConditionalOnMissingBean(ChannelBindingService.class)
```

Comment:
I wonder if we should use SearchStrategy.PARENTS to make this condition stronger.